parquet.go

```diff
@@ -109,7 +110,7 @@ func Open(getReaderFunc GetReaderFunc, columnNames set.StringSet) (*File, error)
 }
 
 // Read - reads single record.
-func (file *File) Read() (record map[string]Value, err error) {
+func (file *File) Read() (record []Value, err error) {
```

Comment:
It would be nice to have searching and indexing together i.e. an enhanced map.  Below is a construct
```go
type Record struct {
	indexNameMap map[index]string
	record                map[string]Value
}

func NewRecord(indexNameMap map[int]string) *Record {
}

func (r *Record) Get(name string) (Value, error) {
}

func (r *Record) Set(name string, value Value) {
}

func (r *Record) Range(f func(name string, value Value) bool) error {
	// Calls f() with name/value by index 0 to len(record)-1.
}
```

Reply:
Why do we need to that? Slice satisfies the requirements, and it is also efficient to use it. 

Is there a good reason to do double maps one to track the index and values

Reply:
I would like to extend the usage beyond minio.  However in parquet and partitioned dataset, parallel reads from different files are common usage.  Eventually this will be added.  Hence keeping a Record is much useful and futuristic.

Reply:
Parallel reads are not a problem for even slices. The problem I have with map is inherent issues of lack of order etc and to be able to store another indexing, it just simply is complex.

I would keep Record as slice of values though and add some function names as you mentioned.

Reply:
`indexNameMap` is created only once per open.  As I worked this sort of problem before, I could send a working code precisely without any complexity.

Reply:
Okay that would be great.will close this PR